homeassistant/components/water_heater/__init__.py

```diff
@@ -56,41 +55,52 @@
 STATE_GAS = "gas"
 
 
+class WaterHeaterOperationMode(StrEnum):
+    """Supported operation modes of a heater."""
+
+    OFF = "off"
+    ON = "on"
+    LEGIONELLA_PREVENTION = "legionella_prevention"
+    BOOST = "boost"
+    AWAY = "away"
+
+
+class WaterHeaterCurrentOperation(StrEnum):
+    """Supported current operations of a heater."""
+
+    OFF = "off"
+    IDLE = "idle"
+    HEATING = "heating"
+    LEGIONELLA_PREVENTION = "legionella_prevention"
+    BOOST = "boost"
+
+
 class WaterHeaterEntityFeature(IntEnum):
     """Supported features of the fan entity."""
 
     TARGET_TEMPERATURE = 1
     OPERATION_MODE = 2
-    AWAY_MODE = 4
+    PRESET_MODE = 8
```

Comment:
Since I am looking at IntEnum/IntFlag (#81903) I have wondered why there are gaps in the enum...
I think maybe it would be clearer to keep the old value, but commented out?
```suggestion
    # AWAY_MODE = 4
    PRESET_MODE = 8
```

Reply:
Seems reasonable yes.